src/Jobs/getUsersData.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace BristolSU\UnionCloud\Jobs;
+
+use BristolSU\Support\User\User;
+use GuzzleHttp\Exception\ClientException;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use BristolSU\UnionCloud\UnionCloud\UnionCloud;
+
+class getUsersData implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    protected int $page;
+    protected int $pageCount;
+    protected int $delayBy;
+    protected $repository;
+
+    protected int $requestLimit = 1;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param int $page
+     * @param int $pageCount
+     * @param int $delayBy
+     */
+    public function __construct(int $page, int $pageCount, int $delayBy)
+    {
+        $this->page = $page;
+        $this->pageCount = $pageCount;
+        $this->delayBy = $delayBy;
+
+        $this->requestLimit = config('unioncloud-portal.users_per_minute');
+
+        // Init Repository
+        $this->repository = app(UnionCloud::class);
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $attributes = [
+            'records_per_page' => $this->requestLimit,
+            'page' => $this->page
+        ];
+
+        try {
+            $Users = $this->repository->getAllUsers($attributes, $this->page)->getRawData();
+        } catch (\Exception $e) {
+            if($e instanceof ClientException && $e->getCode() === 403) {
+                $this->error('Failed to reach UC');
+                return;
```

Comment:
Could you also check to see if the rate limit has been reached here? Not sure what the code is, but catching that execption and sending an error message to say the rate limit was met sounds great

Reply:
There doesn't actually seem to be a specific error to do with rate limit reached - https://unioncloudapiv1dev.docs.apiary.io/#introduction/error-codes/1.-generic-request-level-error-codes

Would it just be a 500 error? If so happy to add that in. Would you also like it to retrigger itself but with a 30 seconds wait? Then it should be fine and should get the rest of the data.

Reply:
I thought there was an error, but I've just tried to hit it and I'm not getting any errors at all (assuming it isn't documented in the error codes). Maybe just send an entry to the log with the exception message and code, so at least I'm aware of any teething issues with this